plugins/logs/mask.go

```diff
@@ -0,0 +1,234 @@
+package logs
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/open-policy-agent/opa/util"
+)
+
+type maskOP string
+
+const (
+	maskOPRemove maskOP = "remove"
+	maskOPUpsert maskOP = "upsert"
+)
+
+type maskRule struct {
+	OP             maskOP      `json:"op"`
+	Path           string      `json:"path"`
+	Value          interface{} `json:"value"`
+	escapedParts   []string
+	unEscapedParts []string
+}
+
+func (r maskRule) String() string {
+	return "/" + strings.Join(r.escapedParts, "/")
+}
+
+func newMaskRule(op maskOP, path string, val interface{}) (maskRule, error) {
+	rule := maskRule{}
+
+	if len(path) == 0 {
+		return rule, fmt.Errorf("mask must be non-empty")
+	} else if path[0] != '/' {
+		return rule, fmt.Errorf("mask must be slash-prefixed")
+	}
+
+	parts := strings.Split(path[1:], "/")
+
+	if parts[0] != "input" && parts[0] != "result" {
+		return rule, fmt.Errorf("mask prefix not allowed")
+	}
+
+	unEscapedParts := make([]string, len(parts))
+	escapedParts := make([]string, len(parts))
+	for i := range parts {
+		var err error
+		unEscapedParts[i], err = url.QueryUnescape(parts[i])
+		if err != nil {
+			return rule, err
+		}
+
+		escapedParts[i] = url.QueryEscape(parts[i])
+	}
+
+	rule = maskRule{
+		OP:             op,
+		Path:           path,
+		Value:          val,
+		unEscapedParts: unEscapedParts,
+		escapedParts:   escapedParts,
+	}
+
+	return rule, nil
+}
+
+func (r maskRule) Erase(event *EventV1) {
```

Comment:
Do we still need to call it `Erase`? Could we make this new one `Mask` or `Apply` or something?